sync/src/main/java/io/apicurio/sync/controller/ArtifactController.java

```diff
@@ -0,0 +1,248 @@
+package io.apicurio.sync.controller;
+
+import java.io.ByteArrayInputStream;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.apicurio.registry.rest.client.RegistryClient;
+import io.apicurio.registry.rest.client.exception.ArtifactNotFoundException;
+import io.apicurio.registry.rest.v2.beans.ArtifactMetaData;
+import io.apicurio.registry.rest.v2.beans.EditableMetaData;
+import io.apicurio.registry.rest.v2.beans.IfExists;
+import io.apicurio.registry.rest.v2.beans.UpdateState;
+import io.apicurio.registry.rest.v2.beans.VersionMetaData;
+import io.apicurio.registry.types.ArtifactState;
+import io.apicurio.registry.types.ArtifactType;
+import io.apicurio.sync.api.Artifact;
+import io.apicurio.sync.api.ArtifactSpec;
+import io.apicurio.sync.api.ArtifactStatus;
+import io.javaoperatorsdk.operator.api.Context;
+import io.javaoperatorsdk.operator.api.Controller;
+import io.javaoperatorsdk.operator.api.DeleteControl;
+import io.javaoperatorsdk.operator.api.ResourceController;
+import io.javaoperatorsdk.operator.api.UpdateControl;
+
+@Controller(generationAwareEventProcessing = true, namespaces = Controller.WATCH_CURRENT_NAMESPACE)
+//DelayRegistrationUntil
+public class ArtifactController implements ResourceController<Artifact> {
+
+    private Logger log = LoggerFactory.getLogger(this.getClass());
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Inject
+    RegistryClient registryClient;
+
+    @Override
+    public DeleteControl deleteResource(Artifact resource, Context<Artifact> context) {
+        return DeleteControl.DEFAULT_DELETE;
+    }
+
+    @Override
+    public UpdateControl<Artifact> createOrUpdateResource(Artifact resource, Context<Artifact> context) {
+
+        log.debug("Handling createOrUpdate {}", resource.getMetadata().getName());
+
+//        Optional<CustomResourceEvent> latestArtifactEvent = context.getEvents().getLatestOfType(CustomResourceEvent.class);
+
+        if (resource.getSpec().getContent() == null) {
+            updateArtifactStatus(resource, "Content is not provided");
+            return UpdateControl.updateStatusSubResource(resource);
+        }
+
+        byte[] content = resource.getSpec().getContent().getBytes();
+        //TODO get external content
```

Comment:
This use case, to load external content by a URL, is crucial for my company, as we will reference most of the schemas (e.g. OpenAPI yaml/json) by a git url.